mldsa/sign.h

```diff
@@ -162,7 +162,15 @@ int crypto_sign(uint8_t *sm, size_t *smlen, const uint8_t *m, size_t mlen,
 int crypto_sign_verify_internal(const uint8_t *sig, size_t siglen,
                                 const uint8_t *m, size_t mlen,
                                 const uint8_t *pre, size_t prelen,
-                                const uint8_t *pk, int externalmu);
+                                const uint8_t *pk, int externalmu)
+__contract__(
+  requires(memory_no_alias(sig, siglen))
+  requires(memory_no_alias(m, mlen))
+  requires(memory_no_alias(pre, prelen))
+  requires(memory_no_alias(pk, CRYPTO_PUBLICKEYBYTES))
+  requires(memory_no_alias(externalmu, sizeof(int)))
+  assigns(object_whole(pk))
```

Comment:
That doesn't look right... the public key pointed to by pk is read but not written by this function. The only output is the function's return value, which can be 0 or -1.

Reply:
fixed.